Reimagine the landing page

ling0x opened the pull request:
Landing page completely reimagined following collaborative feedback shared during the Discord thread.

joshka committed:
Make a landing page

j-g00da committed:
docs: update collapse borders recipe for Ratatui 0.30 (#910)

j-g00da committed:
docs: update collapse borders recipe for Ratatui 0.30 (#910)

ling0x committed:
chore: initial create

ling0x committed:
refactor: redesign landing title and taglines

ling0x committed:
refactor: redesign landing page contents

ling0x committed:
refactor: redesign landing page showcases

ling0x committed:
refactor: redesign landing page features and footer

ling0x committed:
refactor: random dots background

ling0x committed:
chore: merge latest changes from main

ling0x committed:
chore: merge latest changes from main

ling0x committed:
chore: merge latest changes from main

ling0x committed:
Revert "chore: merge latest changes from main"

This reverts commit fa93ef068a15821faf6f93693f836321c2ca1af2.

ling0x committed:
Revert "chore: merge latest changes from main"

This reverts commit eea4a1f2e67bb1ab4188d9d81a5c5a2757272321, reversing
changes made to 3d5d48636bb25b5354fea0c68cb992fd6c6f8d85.

ling0x committed:
Revert "chore: merge latest changes from main"

This reverts commit 3d5d48636bb25b5354fea0c68cb992fd6c6f8d85, reversing
changes made to db84dd9aeb230c01bed72fdd4820ff8f2a08c86b.

ling0x committed:
Revert "refactor: random dots background"

This reverts commit db84dd9aeb230c01bed72fdd4820ff8f2a08c86b.

ling0x committed:
Revert "chore: fix npm audit and updating to latest astro and taildwind (#951)"

This reverts commit abc182f47656b2f0b06c26344269f2d8f2ae1c4f.

ling0x committed:
refactor: added random dots background

ling0x committed:
refactor: finetune random dots background size

ling0x committed:
refactor: extra features into its own component

ling0x committed:
Merge branch 'main' into cy/landing-page-v2
Merge the latest updates from main

ling0x committed:
Reapply "chore: fix npm audit and updating to latest astro and taildwind (#951)"

This reverts commit 66000fb0879dfc84dbbfdccac6746fb6b93d27f5.

ling0x committed:
Reapply "chore: merge latest changes from main"

This reverts commit 6ac47d19f5db4ddff8a70ca103238934055c07fc.

ling0x committed:
Reapply "chore: merge latest changes from main"

This reverts commit 97b6b0d0fa09b1b0404f05e1809ce034f238b5f0.

ling0x committed:
Reapply "chore: merge latest changes from main"

This reverts commit 8e45898f0e4cd365c9e0374da61b26b6ae8ed30e.

ling0x committed:
refactor: updated to latest astro and tailwindcss

ling0x committed:
refactor: remove unused plain css from landing page

ling0x committed:
refactor: landing page design enhancement

ling0x committed:
refactor: re-colour background gradients

ling0x committed:
refactor: added a loading animation

ling0x committed:
refactor: added more showcases

ling0x committed:
chore: autoformatting

ling0x committed:
refactor: styling updates

ling0x committed:
refactor: layout updates

ling0x committed:
refactor: added link to showcase gif

ling0x committed:
refactor: showcase layout updates

ling0x committed:
refactor: correct syntax issue

ling0x committed:
refactor: changing showcase order

ling0x committed:
refactor: added tutorial videos

ling0x committed:
refactor: changed logos

ling0x committed:
refactor: changed logos

ling0x committed:
refactor: updated padding around features

ling0x committed:
refactor: features link to docs

ling0x committed:
refactor: making hero section on landing page mobile friendly

ling0x committed:
chore: feature interface

ling0x committed:
chore: feature interface

ling0x committed:
refactor: responsive design for mobile and tablet devices

ling0x committed:
refactor: wording

ling0x committed:
chore: remove unused imports

ling0x committed:
chore: remove unused imports

ling0x committed:
refactor: changed padding around tutorial videos on landing page

ling0x committed:
refactor: removed unused png

ling0x committed:
chore: use alias for links

ling0x committed:
chore: prettier --write

orhun commented on src/assets/hero-dark.png~:
What are the files ending with `~` character? Are these for backup?

ling0x committed:
chore: npx prettier src/tailwind.css --write

orhun commented on src/content/docs/index.mdx:
```diff
@@ -1,105 +1,158 @@
+          <div class="pb-5">
+            <p class="font-bold text-5xl max-2xl:text-4xl max-sm:text-3xl text-black dark:text-white">Cook up <span class="landing-highlight">delicious</span></p>
+            <p class="font-bold text-5xl max-2xl:text-4xl max-sm:text-3xl text-black dark:text-white">terminal user interfaces</p>
+            <p class="text-lg max-sm:text-sm pt-5 leading-8 max-w-[600px]">Ratatui is a Rust library for building rich terminal user interfaces and applications.</p>
```
```suggestion
            <p class="text-lg max-sm:text-sm pt-5 leading-8 max-w-[600px]">Ratatui is a Rust library for building fast, lightweight, and rich terminal user interfaces 👨‍🍳🐀</p>
```

orhun commented on src/components/ScrambleCode.astro:
```diff
@@ -0,0 +1,167 @@
+        "#fff", // property white
+      ];
+
+      const charset = "!<>-_\\/[]{}—=+*^?#________01234567890abcdefghijklmnopqrstuvwxyz";
```
Can we make this cheesy? 🧀 

```suggestion
      const charset = "🧀!✨<>-🌟_/\\[]{}🐭=+*^?#💛________0123456789🍕abcdefghijklmnopqrstuvwxyz🧡";
```

I tried but it didn't display properly for me :/

Cyberphinx replied:
Fixed!

orhun commented on src/utils/stats.ts:
```diff
@@ -0,0 +1,174 @@
+
+export async function getAllStats() {
+  const [github, crates, showcase] = await Promise.all([
+    getGitHubStats("ratatui-org", "ratatui"),
```
```suggestion
    getGitHubStats("ratatui", "ratatui"),
```

orhun commented on src/components/ScrambleText.astro:
Any chance that we could use random "🐀" or "🧀" instead of 0 and 1?

orhun commented on src/components/ShowcaseCards.astro:
```diff
@@ -0,0 +1,173 @@
+  },
+  {
+    title: "rebel-in-the-sky",
+    owner: "alemidev",
```
```suggestion
    owner: "ricott1",
```

orhun commented on src/components/ShowcaseCards.astro:
```diff
@@ -0,0 +1,173 @@
+  {
+    title: "yozefu",
+    owner: "alemidev",
+    repo: "scope-tui",
```
Pls update these

orhun commented on src/components/ShowcaseCards.astro:
```diff
@@ -0,0 +1,173 @@
+    owner: "alemidev",
+    repo: "scope-tui",
+    url: "https://github.com/zaghaghi/openapi-tui",
+    description: "Unlock the power of Open AI APIs",
```
The description is not correct :/

orhun commented on src/components/ShowcaseCards.astro:
```diff
@@ -0,0 +1,173 @@
+  {
+    title: "crossword",
+    owner: "alemidev",
+    repo: "scope-tui",
```
Needs update

orhun commented on src/components/ShowcaseCards.astro:
```diff
@@ -0,0 +1,173 @@
+    owner: "alemidev",
+    repo: "scope-tui",
+    url: "https://github.com/kyu08/fzf-make",
+    description: "A command line tool that executes make target using fuzzy finder",
```
```suggestion
    description: "Execute make target using fuzzy finder",
```

orhun commented on src/components/ShowcaseCards.astro:
```diff
@@ -0,0 +1,173 @@
+    owner: "alemidev",
+    repo: "scope-tui",
+    url: "https://github.com/mrjackwills/oxker/",
+    description: "A simple TUI to view & control docker containers",
```
```suggestion
    description: "View & control docker containers with a TUI",
```

orhun commented on src/components/ShowcaseCards.astro:
```diff
@@ -0,0 +1,173 @@
+  {
+    title: "trippy",
+    owner: "alemidev",
+    repo: "scope-tui",
```
Needs update

orhun commented on src/components/TutorialVideos.astro:
```diff
@@ -0,0 +1,104 @@
+
+  <p class="max-w-[1440px] pb-12 text-lg leading-8 max-xl:text-center max-sm:text-sm">
+    Watch our ‘recipes’ for building features, explore creative twists from other developers, and
+    discussions that stir up fresh ideas.
```
```suggestion
    Watch our "recipes" for learning how to cook up TUIs, get inspired by other developers, and join conversations that spark new ideas.
```

orhun commented on src/components/Features.astro:
```diff
@@ -0,0 +1,66 @@
+    url: "/concepts/layout/",
+    icon: "📐 ",
+    description:
+      "Constraint-based responsive layouts that automatically adapt to any terminal size—think Flexbox, but for the terminal. Build complex nested layouts with horizontal and vertical splits, automatically sized containers, and percentage-based constraints. Your interface looks great whether it's in a tiny tmux pane or a 4K terminal.",
```
```suggestion
      "Your UI looks great whether it's in a tiny tmux pane or a big terminal. Constraint-based responsive layouts that automatically adapt to any terminal size. Build complex nested layouts with horizontal and vertical splits, automatically sized containers, and percentage-based constraints. (Think Flexbox, but for the terminal.)",
```

orhun commented on src/components/Features.astro:
```diff
@@ -0,0 +1,66 @@
+    url: "/concepts/backends/",
+    icon: "🦀",
+    description:
+      "Memory-safe, thread-safe, and type-safe by design. No C dependencies, no runtime exceptions, no undefined behavior. Just pure Rust goodness that compiles to efficient native code. Works seamlessly with Crossterm, Termion, or Termwiz backends—choose the one that fits your needs.",
```
```suggestion
    "Memory-safe, thread-safe, and type-safe by design. No C dependencies, no runtime exceptions, no undefined behavior. Just pure Rust goodness that compiles to efficient native code. Now supports embedded targets with no_std compatibility too!",
```

ling0x committed:
chore: remove automatically generated backup image files

ling0x committed:
Apply suggestions from code review

Co-authored-by: Orhun Parmaksız <orhunparmaksiz@gmail.com>

ling0x committed:
refactor: edit texts and descriptions and reformatting with prettier

Cyberphinx commented on src/content/docs/index.mdx:
```diff
@@ -1,105 +1,158 @@
+              <SocialIcons />
+            </div>
+          </div>
+        </div>
```
Added the random rat and made it just a little bit bigger as well.

ling0x committed:
refactor: replaced 0 and 1 with 🧀 and 🐀

ling0x committed:
refactor: updated code example snippet, added version number and copy button

ling0x committed:
Merge branch 'main' into cy/landing-page-v3

ling0x committed:
refactor: correct syntax

Cyberphinx commented on code/tutorials/quickstart-ratatui/src/main.rs:
Updated the code, included a version tag sourced from Cargo.toml and added functionality to select and copy code.

ling0x committed:
refactor: scramble code with emojis

ling0x committed:
refactor: scramble code with emojis

orhun commented:
Another thing to minify the content: we could add a "show more" button for expanding the list of apps and tutorials. I think that would be better in terms of less scrolling and not tiring the eyes

orhun committed:
docs(highlights): add highlights for 0.30.0 (#902)

🧀

I mostly skipped fixes and only mentioned the most important changes.
Please let me know if I missed something.

I added some todos and would be nice to get help writing/changing those
sections.

I'd also appreciate if someone can run `npm run format` and commit the
result, apparently my svelte compiler is broken or sum idk

P.S. worked on this live: <https://www.youtube.com/watch?v=nucLyibb5NU>

---------

Co-authored-by: ling0x <88593809+ling0x@users.noreply.github.com>
Co-authored-by: Cyberphinx <88593809+Cyberphinx@users.noreply.github.com>

orhun committed:
docs(highlights): add highlights for 0.30.0 (#902)

🧀

I mostly skipped fixes and only mentioned the most important changes.
Please let me know if I missed something.

I added some todos and would be nice to get help writing/changing those
sections.

I'd also appreciate if someone can run `npm run format` and commit the
result, apparently my svelte compiler is broken or sum idk

P.S. worked on this live: <https://www.youtube.com/watch?v=nucLyibb5NU>

---------

Co-authored-by: ling0x <88593809+ling0x@users.noreply.github.com>
Co-authored-by: Cyberphinx <88593809+Cyberphinx@users.noreply.github.com>

ling0x committed:
Update src/components/ShowcaseCards.astro

Co-authored-by: Orhun Parmaksız <orhunparmaksiz@gmail.com>

ling0x committed:
fix: link to tutorials

ling0x commented:
> Can you maybe change the target branch of this PR to `ratatui-0.30` instead of `main`? (since it will go live with the new release)

Yes that makes sense. Changed!

ling0x committed:
refactor: add Octokit for authenticated GitHub stats, remove unused Svelte formatter

ling0x commented:
> Looks dope!
> 
> Some general comments:
> 
>     * The padding at top seems to be a bit large. We can reduce it so we have less space.
> 
>     * The animation is a bit slow on my machine. But that's probably fine because we are thinking of replacing that with tachyonfx.
> 
>     * The formatting failed on CI
> 
>     * We could show the stars count for the showcase projects (should be dynamically fetched)

- Reduced padding at the top for both desktop and mobile views
- Acknowledged that lower laptop CPU clock speeds may contribute to JavaScript animation lag, but won't optimize further since this will be replaced with tachyonfx
- Ran `npm run format` and removed svelte-formatter as the project contains no Svelte files
- Added GitHub authentication to npm-ci workflow and displayed star counts for all showcases. Successfully tested fetching GitHub stats locally using a PAT in .env, but need to verify the GITHUB_TOKEN works in the npm-ci workflow. The current staging site at https://ratatui.ling.ooo is self-hosted in a Docker container. You are using cloudlfare pages, right? I probably should try host it on cloudflare pages just to replicate your deployment setup to ensure everything works as expected.

ling0x committed:
Merge branch 'ratatui-0.30' into cy/landing-page-v3

ling0x committed:
refactor: reduce the number of showcases

ling0x committed:
fix: fixed typescript syntax errors

orhun commented:
> Added GitHub authentication to npm-ci workflow and displayed star counts for all showcases. Successfully tested fetching GitHub stats locally using a PAT in .env, but need to verify the GITHUB_TOKEN works in the npm-ci workflow. The current staging site at https://ratatui.ling.ooo/ is self-hosted in a Docker container. You are using cloudlfare pages, right? I probably should try host it on cloudflare pages just to replicate your deployment setup to ensure everything works as expected.

Yup. I think it's going to be fine, I can take a look at this later.

> I've trimmed the showcase list to 5 items. Users can click "Explore the apps ::<>" to view more, for example.

Fair!

joshka commented:
This looks sick! Doing a bit of a review now.

joshka commented on src/assets/netflix.png:
Double check the branding guidelines - we probably shouldn't put this in a circle as it looks off.
https://brand.netflix.com/en/assets/logos
We might want to use the entire wordmark here

joshka commented on src/assets/openai.svg:
Might be nice to use OpenAI's wordmark here too rather than the blossom
https://openai.com/brand/

joshka commented on src/assets/ovh.png:
Same (logo with the word): https://corporate.ovhcloud.com/en/newsroom/assets/ (scroll to the end)

joshka commented on src/components/CompanyLogos.astro:
```diff
@@ -0,0 +1,28 @@
+---
+import netflix from "@assets/netflix.png";
+import openai from "@assets/openai.svg";
+import ovhcloud from "@assets/ovh.png";
```
I think there's also a facebook tool for doing network routing type things.

joshka commented on src/components/ShowcaseCards.astro:
```diff
@@ -0,0 +1,121 @@
+import ShowcaseCard from "./ShowcaseCard.astro";
+import type { ShowcaseProject } from "../models/project.ts";
+
+const projects: ShowcaseProject[] = [
```
I'd love to expand this a bit more to the other things in the current app showcase if possible (but also, this being so good makes the app showcase look bad :D)

ling0x replied:
During @orhun's YouTube livestream, some viewers mentioned that there were too many showcases, so we reduced them to five. However, that felt a bit too few, so I’ve added three more back in. The reason the grid makes the app showcase looked bad was that the grid layout cropped the GIFs slightly. I’ve now handcrafted the grid sizes so that each GIF fits properly and looks better. The grid is no longer randomized.

joshka commented on src/components/ShowcaseCards.astro:
```diff
@@ -0,0 +1,121 @@
+    class="pt-[80px] pb-6 text-4xl font-bold text-black max-xl:text-center max-sm:pt-[50px] dark:text-white"
+  >
+    Built with <span class="text-blue-700 dark:text-yellow-500">Ratatui</span> 👨‍🍳🐀
+  </h2>
```
This section and the following sections going full width without a border seems a bit inconsistent with the main part above. What about keeping the margins perhaps?

ling0x replied:
Do you mean to add a border or gradient background around the entire showcase app grid section, or just the text "Build with Ratatui"?

joshka reviewed (approved):
I love so much about this. All my comments are really super nit picky type things. You've done a great job with this. I really like it a lot

ling0x committed:
refactor: relink the feature

ling0x committed:
refactor: handcraft showcase grid display ratios

ling0x committed:
refactor: repalced rat, cheese, chef and square emojis to be like archlinux

ling0x commented on src/components/RandomDots.astro:
Replaced them as images instead of emojis. Looks fine on archlinux. Will test on windows later.

ling0x replied:
Ok, so I've tested the site on a windows machine and the rat, chef, cheese and square images looks the same as on archlinux.

orhun commented:
I just showed the new landing page to a developer friend, he asked where the light theme switch is 👀 
Can we add it next to the social media buttons?

He also pointed out:

- We can improve the SEO by addressing the points raised in https://pagespeed.web.dev
- He really liked the new website :)

ling0x committed:
chore: fixing CI check and formatting issues

ling0x committed:
refactor: keeping margins consistent on laptop device width and below

ling0x committed:
refactor: adjust alignment of chef and mouse images

ling0x committed:
refactor: adjust alignment of chef and mouse images

ling0x committed:
refactor: optimize title for mobile

ling0x committed:
refactor: optimize title for mobile

ling0x committed:
refactor: add theme switch, adjust background colour, improve alignment

ling0x committed:
chore: merge from ratatui-0.30

ling0x committed:
chore: ensure path stability with trailing slash

ling0x committed:
refactor: fix cosmetic details

ling0x committed:
chore: npm run format

ling0x committed:
refactor: showcase cards layout on mobile

ling0x committed:
refactor: showcase card text to image alignment

ling0x committed:
refactor: showcase card text alignment

ling0x committed:
refactor: replace gradient with dark slate background in showcases and videos for better focus

ling0x committed:
feat: added companies logos following their branding guidelines

ling0x committed:
fix: typescript syntax errors and warnings

joshka commented:
Looks great on an iPhone.

Typo - OpenAI-tui is openapi-tui 

Ea logo looks odd.

Same with gitbutler - https://github.com/gitbutlerapp/gitbutler-brand-assets/tree/main/logos/full-logo has the goods

ling0x committed:
chore: npm run format

ling0x committed:
refactor: add links to company logos

ling0x committed:
refactor: replace youtube video on landing page

ling0x committed:
chore: npm run format

ling0x committed:
refactor: use original gitbutler logo

ling0x committed:
refactor: use official electronic arts logo

ling0x committed:
refactor: fix font color

joshka commented:
I agree with @j-g00da here. It's too busy now.
I'm not a fan of any animations (rat scramble / code) - I prefer calm, but happy to defer to the others on this one.

The borders here are unnecessary and add to the busy-ness:
<img width="1097" height="79" alt="image" src="https://github.com/user-attachments/assets/91b62bf3-632d-4c06-b9cb-c5a201a1a6b0" />

I added a filter on the dots to filter it down a bit `filter: brightness(0.4);` and it helped a bit

A good question to ask when looking at any particular part of this page is "where do my eyes get draw to due to color / size / background / positioning?" or "what's the strongest element on this screen and how is it differentiated from the next". Here that answer is everywhere, and it's difficult to tell.

ling0x committed:
refactor: reduce busy-ness

ling0x committed:
refactor: remove unused code

ling0x committed:
refactor: adjust spacing

joshka commented on src/components/ShowcaseCards.astro:
```diff
@@ -0,0 +1,157 @@
+  <div class="m-auto flex max-w-[1440px] flex-wrap gap-6">
+    {
+      projects.map((project: ShowcaseProject, index: number) => (
+        <div
```
I'd suggest pushing this div down into the card so that the the styling is there as a single element rather than two elements - pass the tileData[index] === 1 bit as a flag to the compoent instead.

Same with the vids

ling0x committed:
fix: cargo clippy error

ling0x commented on src/components/StatsDisplay.astro:
```diff
@@ -0,0 +1,59 @@
+  console.log("Fetching stats at build time...");
+  stats = await getAllStats("ratatui", "ratatui");
+  console.log("Successfully fetched stats:", stats);
+} catch (error) {
```
For local development, a `.env` file is used, where `GITHUB_TOKEN` is set to the developer’s personal access token (for example).

ling0x committed:
refactor: consolidate showcase card elements into one component

ling0x commented:
@joshka From a design standpoint, keeping the boxes and gradient really helps maintain the site’s visual structure and sense of depth. The gradient in the stats section gives a nice smooth transition in dark mode, setting a clear visual tone right from the start. Having each showcase framed with its own background makes the content easier to focus on and gives it that gallery feel where every piece stands out on its own. Without these details, the layout starts to feel a bit flat and basic, losing some of that contrast and polish that makes it look refined.

ling0x committed:
refactor: add background to stats

ling0x committed:
refactor: adjust padding in stats on mobile

ling0x committed:
Merge branch 'ratatui-0.30' into cy/landing-page-v3

ling0x committed:
chore: fix cargo clippy errors in lock file

ling0x committed:
refactor: mute company logos to grayscale and reposition them outside the stats banner for a cleaner layout

ling0x committed:
refactor: replace youtube embeds with youtube-nocookie

ling0x committed:
refactor: add alt to all imgs

ling0x committed:
refactor: track svg, webp and xcf files in LFS

ling0x committed:
refactor: include svg, webp and xcf LFS tracking in CONTRIBUTING.md

ling0x committed:
refactor: add aria-label to anchor links for accessibility

ling0x committed:
refactor: use youtube-nocookie, remove error log to browser console

ling0x commented:
https://pagespeed.web.dev/analysis/https-ratatui-ling-ooo/ywnhi1gzsh?form_factor=mobile
Accessibility: 100
Best practice: 92
SEO: 92
The robots.txt is cloudflare managed, so will ignore the robots.txt error. Lighthouse flags Content-Signal as an "unknown directive" because it is an experimental, non-standard robots.txt extension from Cloudflare rather than part of the official spec. Despite the warning, cf reports no negative effect on crawling or search performance.

Next will look at LCP for potential improvements (mostly related to image delivery I believe).

ling0x committed:
refactor: track svg xcf and webp files with Git LFS
